### ISSUE TITLE
All events are sent to the same partition

### ISSUE THREAD

**Issue by alepez** [Azure/azure-event-hubs-c#14]
I have an EventHub with 16 partitions. With the *C Client* all events are going to the partition 14.

You can test it with any of the sample: `send`, `send_batch` and `send_async`. All events are going to the same partition.

I've tested with the nodejs sender sample and it is working as expected. I've tested it with SBExplorer and a nodejs receiver.

I've written a simple nodejs receiver (based on the nodejs sample) to print the partitionId and the partitionKey. You can see [the source here](https://gist.github.com/alepez/2d0621f31677f37f7752d1bc2a2681b5)

This is what happens in the receiver when sending with the *NodeJS Client*: 

```
partitionId: 14, partitionKey: 1
partitionId: 0, partitionKey: 2
partitionId: 3, partitionKey: 3
partitionId: 1, partitionKey: 4
partitionId: 8, partitionKey: 5
partitionId: 5, partitionKey: 6
partitionId: 14, partitionKey: 7
partitionId: 8, partitionKey: 8
partitionId: 4, partitionKey: 9
partitionId: 4, partitionKey: 10
partitionId: 15, partitionKey: 11
partitionId: 4, partitionKey: 12
partitionId: 7, partitionKey: 13
partitionId: 5, partitionKey: 14
partitionId: 5, partitionKey: 15
partitionId: 4, partitionKey: 16

```

With the NodeJS client it works as expected, it is using all partitions.

This is what happens in the receiver when sending with the *C Client*:

```
partitionId: 14, partitionKey: 1
partitionId: 14, partitionKey: 2
partitionId: 14, partitionKey: 3
partitionId: 14, partitionKey: 4
partitionId: 14, partitionKey: 5
partitionId: 14, partitionKey: 6
partitionId: 14, partitionKey: 7
partitionId: 14, partitionKey: 8
partitionId: 14, partitionKey: 9
partitionId: 14, partitionKey: 10
partitionId: 14, partitionKey: 11
partitionId: 14, partitionKey: 12
partitionId: 14, partitionKey: 13
partitionId: 14, partitionKey: 14
partitionId: 14, partitionKey: 15
partitionId: 14, partitionKey: 16
```

All events are going to the partition 14. Note that the partitionKey is the expected one, so the client is sending the correct partitionKey, but to the wrong partition.

I've some suspect this issue is related to #13 because if i send something from the *NodeJS Client* with partitionKey="sender", it goes to partition 14 (I'm guessing the hash of the string "sender" is 14).

**Comment by dcristoloveanu** [comment]
@alepez 
This seems related to #13 indeed. WIll have a look at #13 and we'll see if this fixes this one too.

Thanks,
/Dan

**Comment by alepez** [comment]
@dcristoloveanu it's surely related, but there's a difference. When receiver receives a batch, the partitionKey is always `sender` and it is in partition 14 (of 16). But when it receives a single event, the partitionKey is the expected one. Nevertheless, events always arrives in partition 14.

**Comment by dcristoloveanu** [comment]
@alepez 

Indeed what you observed is consistent with what I also observed while fixing #13.
For one event being sent, if the partitionKey is set on the event then it would go to the proper partition.
If one would send batches, then the partition would always be 'sender'.

This is fixed now with the fix for #13.
Let me know in case you still have issues and I will reopen the issue and assist you.

Thanks,
/Dan
